module/state_synchronization/indexer/collection_executed_metric.go

```diff
@@ -25,6 +26,9 @@ type CollectionExecutedMetricImpl struct {
 
 	collections storage.Collections
 	blocks      storage.Blocks
+
+	blockTransactions map[flow.Identifier][]flow.Identifier // Map to track transactions for each block for sealed metrics
```

Comment:
instead of a vanilla map, how about using a mempool `stdmap.IdentifierMap`. this has the added benefit that it has a size limit so memory is bounded, and has embedded locking